pkg/daemon/cleanup/cleanup.go

```diff
@@ -98,6 +103,14 @@ func removeServices(svcMgr manager.Manager, services []servicescm.Service) error
 			failedRemovals = append(failedRemovals, err)
 		} else {
 			servicesRemoved = append(servicesRemoved, service.Name)
+			backingProcess, err := processName(service.Command)
+			if err != nil {
+				klog.Warningf("could not determine process associated with the %s service", service.Name)
+				continue
+			}
+			if err = waitForProcessToStop(backingProcess); err != nil {
```

Comment:
Should you try killing the process if it doesn't stop?

Reply:
I was on the fence, but I've added that now.